src/internal/pachsql/db.go

```diff
@@ -58,6 +92,85 @@ func OpenURL(u URL, password string) (*DB, error) {
 	return res, errors.EnsureStack(err)
 }
 
+// ListTables returns an array of SchemaTable structs that represent the tables.
+func ListTables(ctx context.Context, db *DB) ([]SchemaTable, error) {
+	rows, err := db.QueryContext(ctx, listTablesQuery)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not list tables")
+	}
+	defer rows.Close()
+	var tables []SchemaTable
+	if err := sqlx.StructScan(rows, &tables); err != nil {
```

Comment:
I think sqlx.SelectContext will handle iterating over all of the rows and struct-scanning into each individual slice element.